mantaray/marshal.go

```diff
@@ -240,6 +258,59 @@ func (n *Node) UnmarshalBinary(data []byte) error {
 			offset += nodeForkPreReferenceSize + refBytesSize
 			return nil
 		})
+	} else if bytes.Equal(versionHash, version02HashBytes) {
+
+		refBytesSize := int(data[nodeHeaderSize-1])
+
+		n.entry = append([]byte{}, data[nodeHeaderSize:nodeHeaderSize+refBytesSize]...)
+		offset := nodeHeaderSize + refBytesSize // skip entry
+		n.forks = make(map[byte]*fork)
+		bb := &bitsForBytes{}
+		bb.fromBytes(data[offset:])
+		offset += 32 // skip forks
+		return bb.iter(func(b byte) error {
+			f := &fork{}
+
+			if len(data) < offset+nodeForkTypeBytesSize {
+				err := fmt.Errorf("not enough bytes for node fork: %d (%d)", (len(data) - offset), (nodeForkTypeBytesSize))
+				return fmt.Errorf("%w on byte '%x'", err, []byte{b})
+			}
+
+			nodeType := uint8(data[offset])
+
+			nodeForkSize := nodeForkPreReferenceSize + refBytesSize
+
+			if nodeTypeIsWithMetadataType(nodeType) {
+				if len(data) < offset+nodeForkPreReferenceSize+refBytesSize+nodeForkMetadataBytesSize {
+					err := fmt.Errorf("not enough bytes for node fork: %d (%d)", (len(data) - offset), (nodeForkPreReferenceSize + refBytesSize + nodeForkMetadataBytesSize))
+					return fmt.Errorf("%w on byte '%x'", err, []byte{b})
+				}
+
+				metadataBytesSize := binary.BigEndian.Uint16(data[offset+nodeForkSize : offset+nodeForkSize+nodeForkMetadataBytesSize])
+
+				nodeForkSize += nodeForkMetadataBytesSize
+				nodeForkSize += int(metadataBytesSize)
+
+				err := f.fromBytes02(data[offset:offset+nodeForkSize], refBytesSize, int(metadataBytesSize))
+				if err != nil {
+					return fmt.Errorf("%w on byte '%x'", err, []byte{b})
+				}
+			} else {
+				if len(data) < offset+nodeForkPreReferenceSize+refBytesSize {
+					err := fmt.Errorf("not enough bytes for node fork: %d (%d)", (len(data) - offset), (nodeForkPreReferenceSize + refBytesSize))
+					return fmt.Errorf("%w on byte '%x'", err, []byte{b})
```

Comment:
Not sure if two fmt.Errorf calls are needed here.

Reply:
I left that on 2 places as I had same line on 4 places total. It seemed "cleaner".

Will streamline it now.